src/ctap2/commands/make_credentials.rs

```diff
@@ -187,15 +188,34 @@ impl PinUvAuthCommand for MakeCredentials {
         }
     }
 
-    fn set_discouraged_uv_option(&mut self) {
-        // "[..] the Relying Party wants to create a non-discoverable credential and not require user verification
-        // (e.g., by setting options.authenticatorSelection.userVerification to "discouraged" in the WebAuthn API),
-        // the platform invokes the authenticatorMakeCredential operation using the marshalled input parameters along
-        // with the "uv" option key set to false and terminate these steps."
-        // Note: This is basically a no-op right now, since we use `get_uv_option() == Some(false)`, to determine if
-        //       the RP is discouraging UV. But we may change that part of the API in the future, so better to be
-        //       explicit here.
-        self.set_uv_option(Some(false))
+    fn handle_discouraged_uv_option(&mut self, info: &AuthenticatorInfo) -> bool {
+        // TODO(MS): Handle here the case where we NEED a UV, the device supports PINs, but hasn't set a PIN.
+        //           For this, the user has to be prompted to set a PIN first (see https://github.com/mozilla/authenticator-rs/issues/223)
+
+        let supports_uv = info.options.user_verification == Some(true);
+        let pin_configured = info.options.client_pin == Some(true);
+        let make_cred_uv_not_required = info.options.make_cred_uv_not_rqd == Some(true);
```

Comment:
We also need to check the `alwaysUv` option from `info` and the `rk` option of the request. The `alwaysUv` option takes precedence over `makeCredUvNotRqd`. And the `makeCredUvNotRqd` option is only supposed to be used in the `rk = false` case.

Also if the maximum supported version from `info.versions` is `FIDO_2_0` then we always need UV for make cred.